### ISSUE TITLE
Have Sentry know about git commit

### ISSUE THREAD

**Issue by exrhizo** [coordinape/coordinape#413]
There is a way to get sentry set up so that when there is a deploy and new bugs start happening, sentry will automatically detect the likely commit.

A lead is that vercel has this env variable:
VERCEL_GIT_COMMIT_SHA

It may involve something with the deploy hooks as well.

**Comment by topocount** [comment]
@reeserj :wave:

**Comment by topocount** [comment]
I've done a bit of research and here are my findings:

**TL;DR** The easiest solution is a [github-level integration](https://docs.sentry.io/product/integrations/source-code-mgmt/github/#configure) [with sentry](https://docs.sentry.io/product/releases/setup/release-automation/)

If we want to tie commits in with specific github releases, we'd want the Sentry Github Action, but that's unnecessary, given that Vercel is used for production deploys and releases don't seem to be cut from Github in this repo.

# WebPack-level integration - not recommended
I played around with the Vercel integration on its own and realized fairly quickly that it doesn't do much at all with the with the `VERCEL_GITHUB_COMMIT_SHA`. This only seems to be of use with a GitHub integration, since sentry seems to then dereference that SHA and pull into commit metadata and other context from the git tree. 

In order to avoid setting up another integration with Sentry, I went on to try out the Webpack sentry integration and got it working fairly easily locally, where compiling the code locally with webpack and triggering an error in the browser was enabling the release and commit tracking functionality in sentry. However, I ran into the following issues:
- the sentry auth token provided to vercel is not authorized to call the set-commit webhook. This would be okay if we could set a project-level auth token, but unfortunately, all manually generated auth tokens must be user-level. Additionally, we'd need to write some code in `craco.config.js` to pull in this user-assigned token from vercel's secret storage. This would be easy enough but the security model is less than great.
- the devX of hooking sentry into webpack is likely not great. We'd probably want to dynamically generate our webpack config based on environment to only include the Sentry webpack plugin in a CI environment in order to avoid annoying build issues. This is simple enough, but its a bit more config to maintain and is susceptible to the security issues above. If the plugin is invoked for local users, they'd need a sentry API key, and the sentry data would be polluted with errors from development work. I'm honestly surprised there isn't a config for this built into the plugin, where it only "phones home" in a CI environment.

**Comment by topocount** [comment]
Update: After testing out the Vercel integration with Sentry it seems possible that it's broken. I say this because Sentry appears to be looking for `VERCEL_GITHUB_COMMIT_SHA` according to their docs, while VERCEL only provides `VERCEL_GIT_COMMIT_SHA` and `VERCEL_GIT_PROVIDER` now.

[Relevant Vercel Docs](https://vercel.com/docs/concepts/projects/environment-variables#system-environment-variables)

[Relevant Sentry Docs](https://docs.sentry.io/product/integrations/deployment/vercel/#project-linking)

**Comment by topocount** [comment]
I'll proceed with plan C, which is a custom script that handles this process via the CLI. I found a "Custom Integration Auth Token" UI that is apparently kosher. for this type of setup.

**Comment by topocount** [comment]
I figured out the step I was missing: We need a corresponding release tag in the frontend. This can be populated in the `Sentry.Init` hook or via the webpack plugin.

The release tag in the frontend is how a deploy is associated with a release in sentry, which matches up with deployed commits 1:1 in the default sentry configuration.
